lib/assets/javascripts/cartodb/table/mapview.js

```diff
@@ -253,10 +253,34 @@ cdb.admin.MapTab = cdb.core.View.extend({
     this.map.layers.bind('remove reset',   this._addLegends, this);
     this.map.layers.bind('remove reset',   this._addTimeline, this);
 
+    this._addLegendBindings();
+
     _.bindAll(this, 'showNoGeoRefWarning', "_exportImage");
 
   },
 
+  _addLegendBindings: function () {
+    var self = this;
+    this.map.layers.each($.proxy(this._bindLegendChange, this));
+    this.map.layers.bind('add', $.proxy(this._bindLegendChange, this));
```

Comment:
using `this._bindLegendChange.bind(this)` is safe here. Anyways I'd use underscore `_.bind`, we try to use jquery as less as possible for utilities